layers/+tools/translate/packages.el

```diff
@@ -0,0 +1,79 @@
+;;; packages.el --- Translate Layer packages File for Spacemacs
+;;
+;; Copyright (c) 2012-2021 Sylvain Benner & Contributors
+;;
+;; Author: Ray Wang <rayw.public@gmail.com>
+;; URL: https://github.com/syl20bnr/spacemacs
+;;
+;; This file is not part of GNU Emacs.
+;;
+;; This program is free software; you can redistribute it and/or modify
+;; it under the terms of the GNU General Public License as published by
+;; the Free Software Foundation, either version 3 of the License, or
+;; (at your option) any later version.
+;;
+;; This program is distributed in the hope that it will be useful,
+;; but WITHOUT ANY WARRANTY; without even the implied warranty of
+;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+;; GNU General Public License for more details.
+;;
+;; You should have received a copy of the GNU General Public License
+;; along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+(defconst translate-packages
+  '(
+    (translate-mode :location (recipe :fetcher github :repo "rayw000/translate-mode"))
+    go-translate
+    popup
+    posframe
+    ))
+
+(defun translate/init-translate-mode ()
+  "Initialize required packages."
+  (use-package translate-mode
+    :demand t
+    :hook (translate-mode . set-translate-mode-paragraph-functions)))
+
+(defun translate/init-go-translate ()
+  (use-package go-translate
+    :demand t
+    :config
+    (progn
+      (defclass spacemacs//original-paragraph-texter (gts-texter) ())
+      (cl-defmethod gts-text ((_ spacemacs//original-paragraph-texter))
+        (translate-get-original-paragraph-text-at-point))
+      (defclass spacemacs//original-paragraph-picker (gts-picker)
+        ((texter :initarg :texter :initform (spacemacs//original-paragraph-texter))))
+      (cl-defmethod gts-pick ((o spacemacs//original-paragraph-picker))
+        (let ((text (gts-text (oref o texter))))
+          (when (= 0 (length (if text (string-trim text) "")))
+            (user-error "Make sure there is any word at point, or selection exists"))
+          (let ((path (gts-path o text)))
+            (cl-values text path))))
+      ;; TODO Popup-el based render
+      ;; (defclass spacemacs//inplace-render (gts-render) ())
+      ;; (cl-defmethod gts-out ((_ spacemacs//inplace-render) task)
+      ;;   (with-slots (ecode result) task
+      ;;     (if ecode
+      ;;         (user-error "%s" result)
+      ;;       (popup-menu* (split-string result "\n")))))
+      (defvar paragraph-translator
```

Comment:
if these two variables are not mutable, defconst are preferred.

also they should go to config.el, with correct naming.